src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/JSSynchronizationContext.cs

```diff
@@ -49,6 +49,22 @@ public static JSSynchronizationContext InstallWebWorkerInterop(bool isMainThread
             var ctx = new JSSynchronizationContext(isMainThread, cancellationToken);
             ctx.previousSynchronizationContext = SynchronizationContext.Current;
             SynchronizationContext.SetSynchronizationContext(ctx);
+
+            // FIXME: make this configurable
+            // we could have 3 different modes of this
+            // 1) throwing on UI + JSWebWorker
+            // 2) throwing only on UI - small risk, more convenient.
+            // 3) not throwing at all - quite risky
+            // deadlock scenarios are:
+            // - .Wait for more than 5000ms and deadlock the GC suspend
+            // - .Wait on the Task from HTTP client, on the same thread as the HTTP client needs to resolve the Task/Promise. This could be also chain of promises.
```

Comment:
```suggestion
            // - .Wait on the Task from HTTP client, on the same thread as the HTTP client needs to resolve the Task/Promise. This could be also be a chain of promises.
```